src/netconf_acm.c

```diff
@@ -1535,16 +1568,25 @@ ncac_check_data_read_filter_r(struct lyd_node **first, const char *user)
 void
 ncac_check_data_read_filter(struct lyd_node **data, const char *user)
 {
+    char **groups = NULL;
+    uint32_t group_count;
+
     assert(data);
 
     /* NACM LOCK */
     pthread_mutex_lock(&nacm.lock);
 
+    if (ncac_collect_groups(sr_get_context(np2srv.sr_conn), user, &groups, &group_count)) {
+        goto cleanup;
+    }
+
     if (*data && !ncac_allowed_tree((*data)->schema, user)) {
-        ncac_check_data_read_filter_r(data, user);
+        ncac_check_data_read_filter_r(data, user, groups, group_count);
     }
 
     /* NACM UNLOCK */
+cleanup:
+    ncac_free_groups(groups, group_count);
```

Comment:
Please refactor the following way
```C
cleanup:
    /* NACM UNLOCK */
    pthread_mutex_unlock(&nacm.lock);
    ncac_free_groups(groups, group_count);
```
so that 1) the comment is followed immediately by the unlock and so that unlock occurs as soon as possible.